### PR TITLE
RentStructure and ProtocolParameters

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes
- [ ] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-block/src/protocol.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use packable::{prefix::StringPrefix, Packable};
+
+use crate::{output::RentStructure, Error};
+
+///
```
Comment (samuel-rufi) [inline bee-block/src/protocol.rs]: remove this ?

File: bee-block/src/output/rent.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    }
+}
+
+/// Specifies the current parameters for the byte cost computation.
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct RentStructure {
+    /// Cost in tokens per virtual byte.
+    pub v_byte_cost: u32,
+    /// The weight factor used for key fields in the ouputs.
```
Comment (samuel-rufi) [inline bee-block/src/output/rent.rs]: outputs

File: bee-block/src/output/rent.rs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    pub v_byte_factor_key: u8,
+    /// The weight factor used for data fields in the ouputs.
```
Comment (samuel-rufi) [inline bee-block/src/output/rent.rs]: outputs

**Review by samuel-rufi — CHANGES_REQUESTED** [review]

**Review by samuel-rufi — DISMISSED** [review]

**Review by grtlr — DISMISSED** [review]

File: bee-block/src/output/mod.rs [hunk 4]
```diff
@@ -218,11 +218,11 @@ impl Output {
     }
 
     /// Verifies if a valid storage deposit was made. Each [`Output`] has to have an amount that covers its associated
-    /// byte cost, given by [`ByteCostConfig`].
+    /// byte cost, given by [`RentStructure`].
```
Comment (Thoralf-M) [inline bee-block/src/output/mod.rs:221]: Should this also be rent cost?
```suggestion
    /// rent cost, given by [`RentStructure`].
```

File: bee-block/src/output/nft.rs [hunk 5]
```diff
@@ -47,10 +47,10 @@ impl NftOutputBuilder {
     /// Creates an [`NftOutputBuilder`] with a provided byte cost config.
```
Comment (Thoralf-M) [inline bee-block/src/output/nft.rs]:
```suggestion
    /// Creates an [`NftOutputBuilder`] with a provided rent structure.
```

File: bee-block/src/output/nft.rs [hunk 6]
```diff
@@ -268,8 +268,8 @@ impl NftOutput {
     /// Creates a new [`NftOutput`] with a provided byte cost config.
```
Comment (Thoralf-M) [inline bee-block/src/output/nft.rs]:
```suggestion
    /// Creates a new [`NftOutput`] with a provided rent structure.
```

File: bee-block/src/output/nft.rs [hunk 7]
```diff
@@ -282,10 +282,10 @@ impl NftOutput {
     /// The amount will be set to the minimum storage deposit.
```
Comment (Thoralf-M) [inline bee-block/src/output/nft.rs:282]: Above this line is `byte cost config`

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**